frontend/javascript/code_snippets.js

```diff
@@ -3,6 +3,70 @@ export function enableCodeHighlighting() {
   hljs.highlightAll();
 }
 
+const h1Elements = document.querySelectorAll('h1');
+const h2Elements = document.querySelectorAll('h2');
+const sidebar = document.querySelector('.sidebar');
+sidebar.style.position = 'fixed';
+sidebar.style.top = '100px';
+sidebar.style.right = '50px';
+sidebar.style.width = 'fit-content';
+sidebar.style.maxWidth = '300px';
+sidebar.style.height = '100%';
+function appendSidebarItem(textContent, tagName) {
+  const newItem = document.createElement('p');
+  newItem.textContent = textContent;
+  if(tagName === 'h1') {
+    newItem.classList.add('sidebar-item');
+    newItem.style.color = '#10b981';
+  }
+  if(tagName === 'h2') {
+    newItem.classList.add('sidebar-item-h2');
+    newItem.style.color = 'gray';
+    newItem.style.marginLeft = '20px';
+    newItem.style.cursor = 'pointer';
+  }
+  sidebar.appendChild(newItem);
+}
+
+document.addEventListener('DOMContentLoaded', function() {
```

Comment:
only `index.js` should be adding event handlers for `DOMContentLoaded`. Hook it from there in the existing `DOMContentLoaded` callback.